source/message.html

```diff
@@ -4,24 +4,66 @@
 <TITLE></TITLE>
 <SCRIPT LANGUAGE=javascript>
 <!--
-function escapeKeyPress(e) {
+var args = window.dialogArguments;
+
+function handleKeyPress(e) {
 	e = e || window.event;
 	if(e.keyCode == 27){ // code for escape
 		window.close();
+	} else if (e.altKey && e.keyCode === 67) {  // Code for Alt+c for copy button
+		onCopyButtonPress();
+	}
+}
+
+function onCopyButtonPress() {
+	// Copy code came from http://www.codestore.net/store.nsf/unid/DOMM-4QHQE8/
+	var rng = document.body.createTextRange();
+	rng.moveToElementText(messageID);
+	rng.scrollIntoView();
+	rng.select();
+	var success = rng.execCommand("Copy");
+	rng.collapse(false);
+	rng.select();
+	if (args) {  // Notify the user about the copy result
+		if (success) {
+			alert(args.item('copySuccessfulAlertText'));
+		} else {
+			alert(args.item('copyFailedAlertText'));
+		}
 	}
-};
+}
+
+function onCloseButtonPress() {
+	//window.open('', '_self', '');  // Only enable if window.close() stops working in a future version, may fix.
```

Comment:
can you explain why this may stop working?

Reply:
I would have to dig back into some old research, but I remember coming across something on StackOverflow, that suggested a potential failure mode could exist. I suspect it would only be in the case of MS changing strictness or some other aspect of the Javascript engine of MSHTML, which is unlikely at this point. But it seemed like a harmless note to leave as an improbable just in case.
Happy to take it out if you prefer.

Reply:
It would be great to add the StackOverlow link if you can find it.

Reply:
I have come to question the currentness of this comment after further research. Removed.